hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/OmSnapshotUtils.java

```diff
@@ -137,13 +144,28 @@ public static void createHardLinks(Path dbPath) throws IOException {
     }
   }
 
+  private static boolean isSamePartition(Path fromPathParent, Path toPathParent) throws IOException {
+    try {
+      if (fromPathParent == null || toPathParent == null) {
+        throw new IOException("From path: " + fromPathParent + " or To path: " + toPathParent + " is null");
+      }
+
+      FileStore fromPathStore = Files.getFileStore(fromPathParent);
+      FileStore toPathStore = Files.getFileStore(toPathParent);
+      return fromPathStore.equals(toPathStore);
+    } catch (IOException ex) {
+      throw new IOException("Failed to get the stores, fromPath: " + fromPathParent + " toPath: " + toPathParent, ex);
+    }
```

Comment:
No need to catch and throw the IOException when you have it in the method signature. It is unnecessary wrapping.